src/firefox/index.js

```diff
@@ -0,0 +1,133 @@
+/* @flow */
+import nodeFs from 'fs';
+import path from 'path';
+import defaultFxRunner from 'fx-runner/lib/run';
+import FirefoxProfile from 'firefox-profile';
+import streamToPromise from 'stream-to-promise';
+
+import * as fs from '../util/promised-fs';
+import {onlyErrorsWithCode, WebExtError} from '../errors';
+import {getPrefs as defaultPrefGetter} from './preferences';
+
+
+export const defaultFirefoxEnv = {
+  XPCOM_DEBUG_BREAK: 'stack',
+  NS_TRACE_MALLOC_DISABLE_STACKS: '1',
+};
+
+export function run(
+    profile: FirefoxProfile, {fxRunner=defaultFxRunner}: Object = {}): Promise {
+
+  console.log(`Running Firefox with profile at ${profile.path()}`);
+  return fxRunner(
+    {
+      'binary': null,
+      'binary-args': null,
+      'no-remote': true,
+      'foreground': true,
+      'profile': profile.path(),
+      'env': {
+        ...process.env,
+        ...defaultFirefoxEnv,
+      },
+      'verbose': true,
+    })
+    .then((results) => {
+      return new Promise((resolve) => {
+        let firefox = results.process;
+
+        console.log(`Executing Firefox binary: ${results.binary}`);
+        console.log(`Executing Firefox with args: ${results.args.join(' ')}`);
+
+        firefox.on('error', (error) => {
+          // TODO: show a nice error when it can't find Firefox.
+          // if (/No such file/.test(err) || err.code === 'ENOENT') {
+          console.log(`Firefox error: ${error}`);
+          throw error;
+        });
+
+        firefox.stderr.on('data', (data) => {
+          console.log(`stdout: ${data.toString().trim()}`);
```

Comment:
@kumar303 shouldn't this be stderr? (e.g. ```console.log(`stderr: ...`);```)

why not using console.error here to route these messages to the stderr?